### PR TITLE
feat: add GCFv2 CAL sample

### PR DESCRIPTION
**Pull request by kelsk**
Adds first party Cloud Audit Logging sample for GCF v2.

- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**

### CODE REVIEW

File: functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Logger;
+
+public class LogCloudEvent implements CloudEventsFunction {
+  private static final Logger logger = Logger.getLogger(LogCloudEvent.class.getName());
+
+  @Override
+  public void accept(CloudEvent event) {
+    logger.info("Event: " + event.getId());
+    logger.info("Event Attribute Names: " + event.getAttributeNames().toString());
```
Comment (averikitsch) [inline functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java:33]: Remove these items

File: functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    logger.info("Event Type: " + event.getType());
+    logger.info("Event Subject: " + event.getSubject());
+
+    if (event.getData() != null) {
+      String cloudEventData = new String(event.getData().toBytes(), StandardCharsets.UTF_8);
+
+      Gson gson = new Gson();
+      JsonObject eventData = gson.fromJson(cloudEventData, JsonObject.class);
+      System.out.println(gson.toJson(eventData));
```
Comment (averikitsch) [inline functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java:42]: remove

File: functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      // Print out details from the Cloud Audit Logging entry
+      JsonObject payload = eventData.getAsJsonObject("protoPayload");
+      logger.info("Resource name: " + payload.get("resourceName").toString());
```
Comment (averikitsch) [inline functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java:46]: Add `logger.info("API Method: " + payload.get("methodName").toString());`

File: functions/v2/cloudevent/log-cloudevent/pom.xml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  limitations under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.example.cloud.functions</groupId>
+  <artifactId>functions-cloudevent-log-cloudevent</artifactId>
```
[the remark this answers is not in the document]
Comment (kelsk) [reply functions/v2/cloudevent/log-cloudevent/pom.xml]: done

File: functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java (hunk 1 above)
Comment (kelsk) [reply functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java]: done

File: functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java (hunk 2 above)
Comment (kelsk) [reply functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java]: oops! done

File: functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java (hunk 3 above)
Comment (kelsk) [reply functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java:44]: done

File: functions/v2/cloudevent/log-cloudevent/src/test/java/functions/LogCloudEventTest.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    private static final TestLogHandler logHandler = new TestLogHandler();
+
+    @BeforeClass
+    public static void beforeClass() {
+        logger.addHandler(logHandler);
+    }
+
+    @Test
+    public void functionsLogCloudEvent_shouldLogCloudEvent() throws Exception {
+        JsonObject request = new JsonObject();
```
[the remark this answers is not in the document]
Comment (kelsk) [reply functions/v2/cloudevent/log-cloudevent/src/test/java/functions/LogCloudEventTest.java]: done

File: functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import io.cloudevents.CloudEvent;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Logger;
+
+public class LogCloudEvent implements CloudEventsFunction {
+  private static final Logger logger = Logger.getLogger(LogCloudEvent.class.getName());
+
+  @Override
+  public void accept(CloudEvent event) {
+    logger.info("Event: " + event.getId());
```
[the remark this answers is not in the document]
Comment (kelsk) [reply functions/v2/cloudevent/log-cloudevent/src/main/java/functions/LogCloudEvent.java]: done

**Review by averikitsch — APPROVED** [review]

**Pull request merged**